sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -987,6 +986,11 @@ private boolean isManagementToken(String lockToken) {
         } else if (message.isSettled()) {
             return Mono.error(logger.logExceptionAsError(
                 new IllegalArgumentException("The message has either been deleted or already settled.")));
+        } else if (message.getLockToken() == null) {
+            // message must be a peeked message (or somehow they created a message w/o a lock token)
+            return Mono.error(
+                logger.logExceptionAsError(new UnsupportedOperationException("This operation is not supported for peeked messages. Only messages received using receiveMessages() or receiveMessagesWithContext() in PEEK_LOCK mode can be settled."))
```

Comment:
We can add this in java doc for all the settlement operation (for ex complete, abandon.. etc ) in the client  
 `@throws UnsupportedOperationException` section.

Reply:
Added.